### ISSUE TITLE
per channel samplewise normalization

### ISSUE THREAD

**Issue by eyaler** [fchollet/keras#1793]
per channel samplewise normalization, should allow doing global contrast normalization per rgb channel

**Comment by jmhessel** [comment]
I think something like this is already supported, unless I'm misunderstanding

http://keras.io/preprocessing/image/

**Comment by eyaler** [comment]
Thanks, afaict this is currently done mixing the rgb channels and not per each channel separately.

**Comment by lemuriandezapada** [comment]
Can't you just use a batch normalization layer with the right axis chosen and mode=0?
http://keras.io/layers/normalization/

**Comment by eyaler** [comment]
see here for suggested fix:
https://github.com/fchollet/keras/commit/4ec84541e3a33f906d49a3a58f5d249296a62c76#commitcomment-16513597
@fchollet can you comment on this?

**Comment by dalupus** [comment]
doing this per channel would also fix the bug where it returns all 0's if working with a grayscale image.
